includes/admin/payments/actions.php

```diff
@@ -239,47 +239,33 @@ function edd_update_payment_details( $data = array() ) {
 add_action( 'edd_update_payment_details', 'edd_update_payment_details' );
 
 /**
- * Trigger a Purchase Deletion
- *
- * @since 1.3.4
- * @param array $data Arguments passed
- * @return void
- */
-function edd_trigger_purchase_delete( $data ) {
-	if ( wp_verify_nonce( $data['_wpnonce'], 'edd_payment_nonce' ) ) {
-
-		$payment_id = absint( $data['purchase_id'] );
-
-		if ( ! current_user_can( 'delete_shop_payments', $payment_id ) ) {
-			wp_die( __( 'You do not have permission to edit this payment record', 'easy-digital-downloads' ), __( 'Error', 'easy-digital-downloads' ), array( 'response' => 403 ) );
-		}
-
-		edd_delete_purchase( $payment_id );
-
-		edd_redirect( admin_url( '/edit.php?post_type=download&page=edd-payment-history&edd-message=payment_deleted' ) );
-	}
-}
-add_action( 'edd_delete_payment', 'edd_trigger_purchase_delete' );
-
-/**
- * New in 3.0, destroys an order, and all it's data, and related data.
+ * New in 3.0, permanently destroys an order, and all its data, and related data.
  *
  * @since 3.0
  *
- * @param $data
+ * @param array $data Arguments passed.
  */
 function edd_trigger_destroy_order( $data ) {
+
 	if ( wp_verify_nonce( $data['_wpnonce'], 'edd_payment_nonce' ) ) {
 
 		$payment_id = absint( $data['purchase_id'] );
 
 		if ( ! current_user_can( 'delete_shop_payments', $payment_id ) ) {
-			wp_die( __( 'You do not have permission to edit this payment record', 'easy-digital-downloads' ), __( 'Error', 'easy-digital-downloads' ), array( 'response' => 403 ) );
+			wp_die( esc_html_e( 'You do not have permission to edit this order', 'easy-digital-downloads' ), esc_html_e( 'Error', 'easy-digital-downloads' ), array( 'response' => 403 ) );
```

Comment:
These should both be just `esc_html__`, rather than echoing. I would also maybe go ahead and add a period to the end of the sentence--feel like it looks a little bit funny on the die screen without it, but I know it wasn't there before, so I am fine without it.

Reply:
The missing period was bugging me, too.  :joy: